src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -764,29 +764,39 @@ static function (ChildDefinition $definition, AsPeriodicTask|AsCronTask $attribu
-        });
-        $container->registerAttributeForAutoconfiguration(Embeddable::class, static function (ChildDefinition $definition) {
-            $definition->addExcludeTag('container.excluded.doctrine.embeddable');
-        });
-        $container->registerAttributeForAutoconfiguration(MappedSuperclass::class, static function (ChildDefinition $definition) {
-            $definition->addExcludeTag('container.excluded.doctrine.mapped_superclass');
-        });
-
-        $container->registerAttributeForAutoconfiguration(JsonStreamable::class, static function (ChildDefinition $definition, JsonStreamable $attribute) {
-            $definition->addExcludeTag('json_streamer.streamable', [
-                'object' => $attribute->asObject,
-                'list' => $attribute->asList,
-            ]);
-        });
+        if (method_exists(Definition::class, 'addExcludeTag')) {
+            $container->registerForAutoconfiguration(CompilerPassInterface::class)
+                ->addExcludeTag('container.excluded.compiler_pass');
+            $container->registerForAutoconfiguration(TestCase::class)
+                ->addExcludeTag('container.excluded.test_case');
+            $container->registerAttributeForAutoconfiguration(AsMessage::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.messenger.message');
+            });
+            $container->registerAttributeForAutoconfiguration(Entity::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.doctrine.entity');
+            });
+            $container->registerAttributeForAutoconfiguration(Embeddable::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.doctrine.embeddable');
+            });
+            $container->registerAttributeForAutoconfiguration(MappedSuperclass::class, static function (ChildDefinition $definition) {
+                $definition->addExcludeTag('container.excluded.doctrine.mapped_superclass');
+            });
+
+            $container->registerAttributeForAutoconfiguration(JsonStreamable::class, static function (ChildDefinition $definition, JsonStreamable $attribute) {
+                $definition->addExcludeTag('json_streamer.streamable', [
+                    'object' => $attribute->asObject,
+                    'list' => $attribute->asList,
+                ]);
+            });
+        } else {
+            $container->registerAttributeForAutoconfiguration(JsonStreamable::class, static function (ChildDefinition $definition, JsonStreamable $attribute) {
+                $definition->addTag('json_streamer.streamable', [
+                    'object' => $attribute->asObject,
+                    'list' => $attribute->asList,
+                ]);
+                $definition->addTag('container.excluded');
```

Comment:
alternatively, we could remove the if/else and always do this:
```suggestion
                $definition->addTag('json_streamer.streamable', [
                    'object' => $attribute->asObject,
                    'list' => $attribute->asList,
                ])->addTag('container.excluded')->setAbstract(true);
```

Reply:
I like this idea. 👍